internal/testplanet/planet.go

```diff
@@ -323,6 +343,106 @@ func (planet *Planet) Shutdown() error {
 	return errlist.Err()
 }
 
+// Ping sends a kad ping request to/from all kad nodes to each other
+// NB: kad nodes include bootstrap, satellite, and storagenode type nodes
+func (planet *Planet) Ping(ctx *testcontext.Context) error {
+	pingErrs := permutePeers(planet, func(a, b Peer, labels [2]string) error {
```

Comment:
instead of the permutePeers thing, add a Kademlia service field to `closablePeer` then use a double for loop over `peer` using `errgroup.Go`.

Reply:
awesome! I didn't know either of those things existed.